src/RequestForm.js

```diff
@@ -425,11 +425,125 @@ class RequestForm extends React.Component {
     this.props.onSubmit(data);
   }
 
+  renderActionMenu = ({ onToggle }) => {
+    const {
+      onCancel,
+      request,
+    } = this.props;
+
+    const { item } = (request || {});
+    const isEditForm = (item && item.barcode);
+
+
+    if (!isEditForm) {
+      return (
+        <Button
+          buttonStyle="dropdownItem"
+          id="clickable-cancel-new-request"
+          data-test-cancel-new-request-action
+          onClick={() => {
+            onCancel();
+            onToggle();
+          }}
+        >
+          <Icon icon="times-circle">
+            <FormattedMessage id="ui-requests.newRequest.cancel" />
+          </Icon>
+        </Button>
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <Button
+          onClick={onCancel}
+          buttonStyle="dropdownItem"
+          data-test-cancel-editing
+        >
+          <Icon icon="times-circle">
+            <FormattedMessage id="ui-requests.edit.cancelEditing" />
+          </Icon>
+        </Button>
+        <Button
+          buttonStyle="dropdownItem"
+          id="clickable-cancel-request"
+          data-test-delete-request
+          onClick={() => {
+            this.setState({ isCancellingRequest: true });
+            onToggle();
+          }}
+        >
+          <Icon icon="trash">
+            <FormattedMessage id="ui-requests.edit.deleteRequest" />
+          </Icon>
+        </Button>
+      </React.Fragment>
+    );
+  };
+
+  renderAddRequestFirstMenu = () => (
+    <PaneMenu>
+      <FormattedMessage id="ui-requests.actions.closeNewRequest">
+        {title => (
+          <IconButton
+            onClick={this.props.onCancel}
+            ariaLabel={title}
+            icon="times"
+          />
+        )}
+      </FormattedMessage>
+    </PaneMenu>
+  );
+
+  renderAddRequestLastMenu = () => {
+    const {
+      pristine,
+      submitting,
+    } = this.props;
+
+    const submittingButtonIsDisabled = pristine || submitting;
+
+    return (
+      <PaneMenu>
+        <Button
+          id="clickable-create-request"
+          type="submit"
+          disabled={submittingButtonIsDisabled}
+          marginBottom0
+          buttonStyle="primary paneHeaderNewButton"
+        >
+          <FormattedMessage id="ui-requests.actions.newRequest" />
+        </Button>
+      </PaneMenu>
+    );
+  }
+
+  renderEditRequestLastMenu = () => {
+    const {
+      pristine,
+      submitting,
+    } = this.props;
+
+    const submittingButtonIsDisabled = pristine || submitting;
```

Comment:
There is the same condition check on 504 line. Please move this check to a component method